bookstoreaggregator/books/models.py

```diff
@@ -1,14 +1,53 @@
+from decimal import Decimal
+
 from django.core.exceptions import ValidationError
 from django.core.validators import MaxValueValidator
 from django.db import models
-from django.db.models.signals import m2m_changed, post_save
-from django.dispatch import receiver
+from django.conf import settings
 
 def validate_integer_price(value):
-    if value != int(value):
-        raise ValidationError('The price must be integer')
+    if isinstance(value, float):
+        value = Decimal(value)
+    elif not isinstance(value, Decimal):
+        raise ValidationError('Invalid type for price')
+
+    if value != value.to_integral_value():
+        raise ValidationError('The price must be an integer')
+
+
+class BookManager(models.Manager):
+    def create_book(self, **kwargs):
+        authors_data = kwargs.pop('authors', [])
+        book = self.model(**kwargs)
+        book.full_clean()
+        book.save()
+
+        if not book.tags.exists():
+            default_tag, _created = Tag.objects.get_or_create(name='tag_not_set')
+            book.tags.add(default_tag)
+
+        for author_data in authors_data:
+            Author.objects.create(book=book, **author_data)
```

Comment:
You still do not assign this authors to a book.

Reply:
<img width="1001" alt="Screenshot 2024-09-19 at 17 02 31" src="https://github.com/user-attachments/assets/61ba0bd8-8628-47a0-bf99-26551e95bc50">